### PR TITLE
Add table oci_mysql_channel. Closes #165

### LINKED ISSUE #165: Add table oci_mysql_channel
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_mysql_channel' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_mysql_channel []

PRETEST: tests/oci_mysql_channel

TEST: tests/oci_mysql_channel
Running terraform
oci_core_vcn.named_test_resource: Creating...
oci_core_vcn.named_test_resource: Creation complete after 2s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaa3vsgj6f3qqf3nyelgbf7lonlbr3fbudojjcancbrxhnq]
oci_core_subnet.named_test_resource: Creating...
oci_core_subnet.named_test_resource: Creation complete after 1s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaapf4p2kztvwruqughtkgvw6gjmutntekobo7woaggh3hxf43bs5va]
oci_mysql_mysql_db_system.named_test_resource: Creating...
oci_mysql_mysql_db_system.named_test_resource: Still creating... [10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [1m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [2m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [3m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [4m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [5m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [6m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m10s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m20s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m30s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m40s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [7m50s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Still creating... [8m0s elapsed]
oci_mysql_mysql_db_system.named_test_resource: Creation complete after 8m8s [id=ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3aavrxktbre7u5bubcdzqfqnbpgsbnhwzmo2avkyqypdvi3nxuja]
oci_mysql_channel.named_test_resource: Creating...
oci_mysql_channel.named_test_resource: Still creating... [10s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [20s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [30s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [40s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [50s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m0s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m10s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m20s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m30s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m40s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [1m50s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [2m0s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [2m10s elapsed]
oci_mysql_channel.named_test_resource: Still creating... [2m20s elapsed]
oci_mysql_channel.named_test_resource: Creation complete after 2m27s [id=ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq]

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

description = "steampipetest5806"
region = "ap-mumbai-1"
resource_id = "ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq"
resource_name = "steampipetest5806"
state = "NEEDS_ATTENTION"
target = tolist([
  {
    "applier_username" = "admin"
    "channel_name" = "replication_channel"
    "db_system_id" = "ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3aavrxktbre7u5bubcdzqfqnbpgsbnhwzmo2avkyqypdvi3nxuja"
    "target_type" = "DBSYSTEM"
  },
])
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
time_created = "2021-07-07 10:58:51.624 +0000 UTC"

Running SQL query: test-get-query.sql

[
  {
    "description": "steampipetest5806",
    "display_name": "steampipetest5806",
    "freeform_tags": {
      "name": "steampipetest5806"
    },
    "id": "ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq",
    "is_enabled": true,
    "lifecycle_state": "NEEDS_ATTENTION"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest5806",
    "freeform_tags": {
      "name": "steampipetest5806"
    },
    "id": "ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq",
    "is_enabled": true,
    "lifecycle_state": "NEEDS_ATTENTION"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest5806"
  }
]
✔ PASSED

POSTTEST: tests/oci_mysql_channel

TEARDOWN: tests/oci_mysql_channel

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created
from
  oci_mysql_channel;

+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
| display_name               | id                                                                                              | applier_username | channel_name        | db_system_id                                                                                     | target_type |
+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
| steampipetest6778          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaaoc2vf4oilupxv7qb2kbwzgpakxwkfkb2qh2kz7a4lbi6g6jsjzmq | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaawaugsvyuc4j6zgd6k3vriyxdd5xauv2ul4k73mi7b3ymuexnalmq | DBSYSTEM    |
| steampipetest4743          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaao3okihi45w7cuwmmtyey3k3hoecnfymspsv7tmokel33jcjq4hna | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaawacx7ipm3vqmtaaqpmwhyumzx7kantehdksv7opn7vbeow4iqc6a | DBSYSTEM    |
| mysqlchannel20210707110707 | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaafaakvgx4sqo2oniql7l62hoqwxzcowlrd7agh6buxmnvh66aftcq | test             | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa5i6il6caoayjfkbfe7eeqzple7khcipxmufenxc6zdkhqcz35azq | DBSYSTEM    |
| steampipetest5806          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3aavrxktbre7u5bubcdzqfqnbpgsbnhwzmo2avkyqypdvi3nxuja | DBSYSTEM    |
+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
```

### List disabled channels

```sql
select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created,
  time_updated,
  is_enabled
from
  oci_mysql_channel
where
  not is_enabled;

+----------------------------+-------------------------------------------------------------------------------------------------+-------------+----------+---------------------+---------------------+------------+
| display_name               | id                                                                                              | description | state    | time_created        | time_updated        | is_enabled |
+----------------------------+-------------------------------------------------------------------------------------------------+-------------+----------+---------------------+---------------------+------------+
| mysqlchannel20210707110707 | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaafaakvgx4sqo2oniql7l62hoqwxzcowlrd7agh6buxmnvh66aftcq | testing     | INACTIVE | 2021-07-07 05:38:10 | 2021-07-07 11:01:44 | false      |
+----------------------------+-------------------------------------------------------------------------------------------------+-------------+----------+---------------------+---------------------+------------+
```

### List target details of channels

```sql
select
  display_name,
  id,
  target ->> 'applierUsername' as applier_username,
  target ->> 'channelName' as channel_name,
  target ->> 'dbSystemId' as db_system_id,
  target ->> 'targetType' as target_type
from
  oci_mysql_channel;

+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
| display_name               | id                                                                                              | applier_username | channel_name        | db_system_id                                                                                     | target_type |
+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
| steampipetest4743          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaao3okihi45w7cuwmmtyey3k3hoecnfymspsv7tmokel33jcjq4hna | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaawacx7ipm3vqmtaaqpmwhyumzx7kantehdksv7opn7vbeow4iqc6a | DBSYSTEM    |
| mysqlchannel20210707110707 | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaafaakvgx4sqo2oniql7l62hoqwxzcowlrd7agh6buxmnvh66aftcq | test             | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa5i6il6caoayjfkbfe7eeqzple7khcipxmufenxc6zdkhqcz35azq | DBSYSTEM    |
| steampipetest5806          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaauchqhzqr6ff4sx7p44fh5zhkyla7mycrb7qf3e6b2seruwiq42qq | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaa3aavrxktbre7u5bubcdzqfqnbpgsbnhwzmo2avkyqypdvi3nxuja | DBSYSTEM    |
| steampipetest6778          | ocid1.mysqlchannel.oc1.ap-mumbai-1.aaaaaaaaoc2vf4oilupxv7qb2kbwzgpakxwkfkb2qh2kz7a4lbi6g6jsjzmq | admin            | replication_channel | ocid1.mysqldbsystem.oc1.ap-mumbai-1.aaaaaaaawaugsvyuc4j6zgd6k3vriyxdd5xauv2ul4k73mi7b3ymuexnalmq | DBSYSTEM    |
+----------------------------+-------------------------------------------------------------------------------------------------+------------------+---------------------+--------------------------------------------------------------------------------------------------+-------------+
```
```
</details>

### CODE REVIEW

File: docs/tables/oci_mysql_channel.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+select
+  display_name,
+  id,
+  description,
+  lifecycle_state as state,
+  time_created
+from
+  oci_mysql_channel;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_channel.md:19]:
```suggestion
```

File: docs/tables/oci_mysql_channel.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  lifecycle_state as state,
+  time_created,
+  time_updated,
+  is_enabled
+from
+  oci_mysql_channel
+where
+  not is_enabled;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_channel.md:37]:
```suggestion
```

File: oci/service.go [hunk 3]
```diff
@@ -51,6 +52,7 @@ type session struct {
 	KmsManagementClient            keymanagement.KmsManagementClient
 	KmsVaultClient                 keymanagement.KmsVaultClient
 	LoggingManagementClient        logging.LoggingManagementClient
+	MysqlClient                    mysql.ChannelsClient
```
Comment (bigdatasourav) [inline oci/service.go:55]:
```suggestion
	MysqlChannelClient                    mysql.ChannelsClient
```

File: oci/service.go [hunk 4]
```diff
@@ -801,6 +803,44 @@ func dnsService(ctx context.Context, d *plugin.QueryData) (*session, error) {
 	return sess, nil
 }
 
+// mysqlService returns the service client for OCI Mysql Service
+func mysqlService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```
Comment (bigdatasourav) [inline oci/service.go:807]:
```suggestion
func mysqlChannelService(ctx context.Context, d *plugin.QueryData, region string) (*session, error) {
```

File: oci/table_oci_mysql_channel.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the Channel.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "description",
+				Description: " A user-supplied description of the backup.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_channel.go:43]:
```suggestion
				Description: "A user-supplied description of the backup.",
```

File: oci/table_oci_mysql_channel.go [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+			},
+			{
+				Name:        "time_updated",
+				Description: "The time the Channel was last updated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getMysqlChannel,
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+			{
+				Name:        "is_enabled",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_channel.go:73]: is_enabled should be after state

File: oci/table_oci_mysql_channel.go [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("listMysqlChannels", "Compartment", compartment, "OCI_REGION", region)
+
+	// Create Session
+	session, err := mysqlService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	plugin.Logger(ctx).Trace("1234")
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_channel.go:154]:
```suggestion
```

File: oci/table_oci_mysql_channel.go [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.MysqlClient.ListChannels(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, channel := range response.Items {
+			plugin.Logger(ctx).Trace("List2345", channel)
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_channel.go:173]:
```suggestion
```

File: oci/table_oci_mysql_channel.go [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+		return nil, err
+	}
+
+	return response.Channel, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func channelTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
+
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_channel.go:234]:
```suggestion
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**